ast/compile.go

```diff
@@ -1806,7 +1808,9 @@ func (vis *ruleArgLocalRewriter) Visit(x interface{}) Visitor {
 	switch v := t.Value.(type) {
 	case Var:
 		gv, ok := vis.stack.Declared(v)
-		if !ok {
+		if ok {
+			vis.stack.Seen(v)
```

Comment:
This will only be called when rule args are traversed, and I believe there there is no way of assigning to such vars without getting a `redeclared` error (correct me if I'm wrong). So in essence, incrementing the counter for a seen var here is irrelevant for this feature; but I added this here fore the sake of consistency.

Reply:
Just thinking aloud about how this branch could be triggered, perhaps
```rego
f(x, x) { true }
```
So when the second `x` is visited, it will already have been deal with...?

Reply:
Yes, that would do it. But I don't think there is ever a case that would trigger here that applies to the problem of unused local var assignments (might be wrong, though; at which case I should add a test case for it). But it's good to be consistent, I think. Who knows if this is useful fore something else in the future.